### PR TITLE
feat: http client add sni support

### PR DESCRIPTION
**Pull request by helintongh**
when one server with multiple domain, client should support Server Name Indication can connect to server.

### CODE REVIEW

**Comment by helintongh** [comment]
这个功能做一下说明，现代浏览器发送https请求时默认会加上server name字段。
因此，coro http需要增加这一功能

image

File: include/cinatra/coro_http_client.hpp [hunk 1]
```diff
@@ -1680,6 +1689,7 @@ class coro_http_client {
   std::unique_ptr<asio::ssl::stream<asio::ip::tcp::socket &>> ssl_stream_;
   bool ssl_init_ret_ = true;
   bool use_ssl_ = false;
+  std::string sni_hostname_ = "";
```
Comment (qicosmos) [inline include/cinatra/coro_http_client.hpp:1692]: also add CINATRA_ENABLE_SSL with sni_hostname_?

Comment (helintongh) [reply include/cinatra/coro_http_client.hpp:1692]: yeah, only https need this it's tls extension.

**Comment by qicosmos** [comment]
i will test the code.

**Comment by helintongh** [comment]
beast have same logic code.Test it need domain. Need Someone offer it. 
[beast example](https://github.com/boostorg/beast/blob/bfd4378c133b2eb35277be8b635adb3f1fdaf09d/example/http/client/sync-ssl/http_client_sync_ssl.cpp#L66)

**Review by qicosmos — APPROVED** [review]
LGTM

**Pull request merged**